src/main.js

```diff
@@ -107,11 +128,19 @@ try {
 	console.error(err);
 }
 
-// Set path according to being built or not
+// If root user, use a root-owned directory for user data
+var configDir = path.join(process.env.HOME, product.dataFolderName);
 if (process.env.VSCODE_DEV) {
-	var appData = app.getPath('appData');
-	app.setPath('userData', path.join(appData, 'Code-Development'));
+	configDir += '-dev';
```

Comment:
Do we need this actually? We already have this: https://github.com/Microsoft/vscode/blob/master/src/vs/workbench/electron-main/env.ts#L79